loopback.go

```diff
@@ -480,11 +542,150 @@ var _ fs.NodeRenamer = (*Node)(nil)
 // Rename implements fs.NodeRenamer interface for *Node
 func (n *Node) Rename(ctx context.Context,
 	req *fuse.RenameRequest, newDir fs.Node) (err error) {
+	time.Sleep(latency)
 	np := filepath.Join(newDir.(*Node).realPath, req.NewName)
 	op := filepath.Join(n.realPath, req.OldName)
 	defer func() {
 		log.Printf("%s.Rename(%s->%s): error=%v",
 			n.realPath, op, np, err)
 	}()
+	defer func() {
+		if err == nil {
+			n.fs.moveAllxattrs(ctx, op, np)
+		}
+	}()
 	return os.Rename(op, np)
 }
+
+func xattrNameMutate(name string) string {
+	if strings.HasPrefix(name, a_KAUTH_FILESEC_XATTR) {
+		name = g_KAUTH_FILESEC_XATTR + name[len(g_KAUTH_FILESEC_XATTR):]
+	}
+	return name
+}
+
+var _ fs.NodeGetxattrer = (*Node)(nil)
+
+// Getxattr implements fs.Getxattrer interface for *Node
+func (n *Node) Getxattr(ctx context.Context,
+	req *fuse.GetxattrRequest, resp *fuse.GetxattrResponse) (err error) {
+	time.Sleep(latency)
+	if !inMemoryXattr {
+		return fuse.ENOTSUP
+	}
+
+	defer func() {
+		log.Printf("%s.Getxattr(%s): error=%v", n.realPath, req.Name, err)
+	}()
+	n.fs.xlock.RLock()
+	defer n.fs.xlock.RUnlock()
+	if x := n.fs.xattrs[n.realPath]; x != nil {
+
+		name := xattrNameMutate(req.Name)
+
+		var ok bool
+		resp.Xattr, ok = x[name]
+		if ok {
+			return nil
+		}
+	}
+	return fuse.ENOATTR
+}
+
+var _ fs.NodeListxattrer = (*Node)(nil)
+
+// Listxattr implements fs.Listxattrer interface for *Node
+func (n *Node) Listxattr(ctx context.Context,
+	req *fuse.ListxattrRequest, resp *fuse.ListxattrResponse) (err error) {
+	time.Sleep(latency)
+	if !inMemoryXattr {
+		return fuse.ENOTSUP
+	}
+
+	defer func() {
+		log.Printf("%s.Listxattr(%d,%d): error=%v",
+			n.realPath, req.Position, req.Size, err)
+	}()
+	n.fs.xlock.RLock()
+	defer n.fs.xlock.RUnlock()
+	if x := n.fs.xattrs[n.realPath]; x != nil {
+		names := make([]string, 0)
+		for k := range x {
+			if !strings.HasPrefix(k, g_KAUTH_FILESEC_XATTR) {
```

Comment:
It looks like this xattr is an ACL thing that macOS uses. Presumably osxfuse changes the name so that the underlying file doesn't get the same ACL. So there's no need to change the name in the in-memory xattr. Will update the PR to remove